gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/processor/subscription/SubscriptionProcessor.java

```diff
@@ -166,15 +179,28 @@ private String computeCtxClientIdentifier(final MutableExecutionContext ctx, fin
 
     private String computeHashOrDefault(final String valueToHash, final String fallback) {
         try {
-            MessageDigest md = MessageDigest.getInstance("SHA-256");
-            md.update(valueToHash.getBytes());
-            byte[] digest = md.digest();
-            return Hex.toHexString(digest);
-        } catch (NoSuchAlgorithmException e) {
+            Cache<String, String> hashedValueCache = getOrCreateCache();
+            String hashedValue = hashedValueCache.get(valueToHash);
+            if (hashedValue == null) {
+                MessageDigest md = MessageDigest.getInstance("SHA-256");
```

Comment:
according to this https://www.baeldung.com/sha-256-hashing-java SHA3-256 is faster.

Reply:
Not always but yes good point will update it.